crates/astria-gossipnet/src/network.rs

```diff
@@ -85,33 +92,31 @@ impl NetworkBuilder {
         self
     }
 
-    pub fn build(self) -> Result<Network> {
-        Network::new(self.bootnodes, self.port)
+    pub fn keypair(mut self, keypair: Keypair) -> Self {
+        // TODO: load/store keypair from disk
+        self.keypair = Some(keypair);
+        self
     }
-}
 
-impl Default for NetworkBuilder {
-    fn default() -> Self {
-        Self::new()
+    pub fn with_mdns(mut self, with_mdns: bool) -> Self {
```

Comment:
Please add a doc comment.

very minor nit (feel free to ignore): on a builder it's fine to just call this `fn mdns` or `fn set_mdns`